File: backend/src/main/java/com/digital_banking/backend/entities/BankAccount.java
package com.digital_banking.backend.entities;


import com.digital_banking.backend.enums.AccountStatus;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "TYPE", discriminatorType = DiscriminatorType.STRING, length = 4)
@Getter @Setter
public class BankAccount {
    @Id
    private String id;
    private Date createdAt;
    private double balance;
    @Enumerated(EnumType.STRING)
    private AccountStatus status;
    private String currency;

    @ManyToOne
    private Customer customer;

    @OneToMany(mappedBy = "bankAccount", fetch = FetchType.LAZY)
    private List<AccountOperation> accountOperation;

}

File: client/src/app/core/interfaces/bank-account.ts
import {Customer} from './customer';
import {BankAccountStatus} from '@core/enums/bank-account-status';

export interface BankAccount {
  id ?: number;
  name : string;
  balance : number;
  currency : string;
  status : BankAccountStatus;
  customers : Customer[];
  updated_at : Date;
  created_at : Date;
}
